gobblin-utility/src/main/java/gobblin/util/callbacks/CallbacksDispatcher.java

```diff
@@ -43,6 +45,7 @@
 public class CallbacksDispatcher<L> {
   private final Logger _log;
   private final List<L> _listeners = new ArrayList<>();
+  private final List<WeakReference<L>> _autoListeners = new ArrayList<>();
```

Comment:
It is essentially a set (e.g. values are dummies). The set is useful because listeners might be removed often, which is faster in a set. I just want to avoid reinventing the wheel and possibly making mistakes.